src/pcre2_auto_possess.c

```diff
@@ -456,6 +456,7 @@ switch(c)
        {
        /* Early return if there is not enough space. This should never
        happen, since all clists are shorter than 5 character now. */
+       PCRE2_DEBUG_UNREACHABLE();
```

Comment:
definitely assert

Reply:
in any case, included a little more information that could be used to start a fix for this and that could be also used IMHO as a good example on how to use thess macros.

Reply:
I see that Carlo has added a MAX_LIST variable in GenerateUcd.py, though it doesn't seem to be used yet in that script.

Would adding a check around line 851 that throws an exception if len(s) > (MAX_LIST - 3) be a good idea? Then anyone who is regenerating the property tables will notice if this is a problem.

The reason for comparing len(s) to MAX_LIST - 3 is because two slots in list are used for other purposes, and it needs to be terminated with the special value NOTACHAR, so the number of slots available for codepoints are MAX_LIST - 3.

If this ever happens, simply increasing the value of MAX_LIST in pcre2_auto_possess.c and GenerateUcd.py is all that is needed... then the stack-allocated arrays used to store those values will get bigger and everything will be OK.

I'm just thinking more about this... If we adjust GenerateUcd.py so that it crashes if ever ucd_caseless_sets are too big... do we still need the fail-safe in pcre2_auto_possess.c? Would it be enough to simply assert PCRE2_ASSERT(clist_dest < list + MAX_LIST)?

This is not simply a technical issue, but more of a "project policy" issue.